umbra-js/src/classes/Umbra.ts

```diff
@@ -560,6 +633,26 @@ export class Umbra {
     return stealthFromPrivate.privateKeyHex;
   }
 
+  /**
+   * @notice Helper method to return parsed localOverrides and LookupOverrides
+   * @param overrides Override the gas limit, gas price, nonce, or advanced mode.
+   * When `advanced` is false it looks for public keys in StealthKeyRegistry, and when true it recovers
+   * them from on-chain transaction when true
+   */
+  static parseOverrides(overrides: SendOverrides = {}) {
+    const localOverrides = { ...overrides }; // avoid mutating the object passed in
```

Comment:
We should add some tests for this method to ensure that things are not mutated

Reply:
Hmm although then do we need to export this method? That's kind of annoying, since it would be nice to keep this not exported 🤔

Reply:
Exported and tested here: https://github.com/ScopeLift/umbra-protocol/pull/426/commits/4003eedc816f957152059f7adf6c81ae1665fd65 Maybe once you think the test is solid we can remove the test and the export.

Reply:
> Maybe once you think the test is solid we can remove the test and the export.

I think it's ok to keep the export and just add a comment explaining why (i.e. for tests only, not intended to be removed). If you remove the test/export after testing, then you need to re-write a test and do that process any time that method is changed, which partly defeats the purpose of having tests